service/controller/v19/resource/cloudformation/update.go

```diff
@@ -50,14 +50,27 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange inte
 			force := false
 			shutdown := true
 			wait := true
+			instance := ""
+
 			for _, v := range volumes {
 				for _, a := range v.Attachments {
+					instance = a.InstanceID
 					err := sc.EBSService.DetachVolume(ctx, v.VolumeID, a, force, shutdown, wait)
 					if err != nil {
 						return microerror.Mask(err)
 					}
 				}
 			}
+
+			i := &ec2.TerminateInstancesInput{
+				InstanceIds: []*string{
+					aws.String(instance),
+				},
+			}
+			_, err = sc.AWSClient.EC2.TerminateInstances(i)
+			if err != nil {
+				return microerror.Mask(err)
+			}
```

Comment:
Initializing the termination here makes sense. I never liked the EBS service implementation manipulating the actual instance state. For the style guide it would be awesome to scope it in its own `{ ... }` block.

Reply:
Why?

Reply:
Because the team writes code like that in a lot of places. With the short and simplistic variable names like `i` for input and `o` for output when working with the AWS SDK it is always the same easy way when scoped. You do not need to change the code when it is extended. You do not need to think about other weird names and make the code different everywhere and thus harder to scan.

Reply:
> Because the team writes code like that in a lot of places. 

Thats not an argument :smile: 

I understood you would like to scope the local variables. Makes sense. How do you distinguish between function and scope then? When to use what? (naturally I'd rather make it a function then scoping this inside the function)

Reply:
When you want to have a dedicated function go ahead. Otherwise I would appreciate the scope. There is no "rule" to prefer one over the other.

Reply:
I've refactored this into a dedicated method and also made sure to only terminate one stopped master there. Still not the most beautiful thing. PTAL